src/main/java/macrobase/analysis/stats/FFT.java

```diff
@@ -0,0 +1,66 @@
+package macrobase.analysis.stats;
+
+import macrobase.analysis.pipeline.operator.MBStream;
+import macrobase.analysis.transform.FeatureTransform;
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+import org.apache.commons.math3.complex.Complex;
+import org.apache.commons.math3.linear.ArrayRealVector;
+import org.apache.commons.math3.linear.RealVector;
+import org.apache.commons.math3.transform.DftNormalization;
+import org.apache.commons.math3.transform.FastFourierTransformer;
+import org.apache.commons.math3.transform.TransformType;
+
+import java.util.List;
+
+public class FFT implements FeatureTransform {
+    private RealVector metricVector;
+    private RealVector paddedInput;
+    private RealVector transformedMetricVector;
+    private FastFourierTransformer transformer;
+    private Complex[] FFTOutput;
+    private int nextPowTwo;
+
+    private final MBStream<Datum> output = new MBStream<>();
+
+
+    public FFT(MacroBaseConf conf) {
+        //no config options or anything for now.
+    }
+
+    @Override
+    public void consume(List<Datum> records) throws Exception {
+        for (Datum d: records){
+            metricVector = d.getMetrics();
+            // TODO: look for decent FFT implementation that doesn't need pwr of 2
+            nextPowTwo = Math.max(2,2*Integer.highestOneBit(metricVector.getDimension()-1));
+            paddedInput = metricVector.append(new ArrayRealVector(nextPowTwo - metricVector.getDimension()));
+
+            transformer = new FastFourierTransformer(DftNormalization.STANDARD);
```

Comment:
If the FFT is the bottleneck JTransforms may be faster, https://www.quora.com/What-is-the-fastest-fast-fourier-transform-FFT-library-for-java . It also supports non-pwr-of-2 sizes but padding is often faster than using a weird size. JTransforms means you'll be working with raw double[] rather than nice vectors of complex numbers though haha.